lib/puppet/provider/netapp_net_port/cmode.rb

```diff
@@ -0,0 +1,51 @@
+require_relative '../netapp_cmode'
+
+Puppet::Type.type(:netapp_net_port).provide(:cmode, :parent => Puppet::Provider::NetappCmode) do
+  @doc = "Manage Netapp network port. [Family: cluster]"
+  
+  confine    :feature => :posix
+  defaultfor :feature => :posix
+
+  netapp_commands :netportlist => {:api => 'net-port-get-iter', :iter => true, :result_element => 'attributes-list'}
+  netapp_commands :netportmdfy => 'net-port-modify'
+
+  mk_resource_methods
+
+  def self.instances
+    Puppet.debug("Puppet::Provider::Netapp_net_port.cmode self.instances: Got to self.instances.")
+    results = netportlist() || []
+    netportinfos = []
+
+    results.each do |result|
+      node_name1 = result.child_get_string('node')
+      port_name1 = result.child_get_string('port')
+      node_port_name = node_name1 + "@" + port_name1
+
+      net_port_info_hash = {
+        :node_port_name => node_port_name,
```

Comment:
I know this doesn't make sense, but the `isnamevar` property or parameter in the type (in this case `node_port_name`) should still be returned as just `:name` by the provider `self.instances` method for it to be detected. Otherwise it raises this error:
```
Error: Could not run: No resource and no name in property hash in cmode instance
```
So just change this line to `:name => node_port_name`